bundles/http/Citrix/Slack/2e8a194d-0c01-44c1-9a62-a5ee1d5ecbb3.js

```diff
@@ -72,185 +73,168 @@ integration.define({
-                                channel.ts = favoriteChannel.ts
-                            }
-                        });
+            do {
 
-                        parsedAllChannels.push({
-                            id: channel.id,
-                            user: channel.user,
-                            name: channel.name,
-                            ts: channel.ts,
-                            is_favorite: channel.is_favorite
-                        })
+                const channelMember = channelMembers[indexChannelsMembers]
 
-                        indexChannelsMembers++
-                    } while (channelMembers[indexChannelsMembers] !== undefined)
+                channel.is_favorite = false
+                channel.user = channelMember
+                channel.ts = null
+                parsedAllFavoriteChannels.forEach((favoriteChannel) => {
+                    if (channel.id === favoriteChannel.text && channelMember === favoriteChannel.user) {
+                        channel.is_favorite = true
+                        channel.ts = favoriteChannel.ts
+                    }
+                });
 
-                }
-                indexChannels++
-            } while (allChannels[indexChannels] !== undefined)
+                parsedAllChannels.push({
+                    id: channel.id,
+                    user: channel.user,
+                    name: channel.name,
+                    ts: channel.ts,
+                    is_favorite: channel.is_favorite
+                })
 
-            if (parsedAllChannels) {
-                dataStore.save("favorite_channels_members", parsedAllChannels)
-            }
+                indexChannelsMembers++
+            } while (channelMembers[indexChannelsMembers] !== undefined)
 
-        } else {
-            throw new Error(`Could not retrieve channels (${respChannels.status}: ${respChannels.statusText})`);
         }
+        indexAllChannels++
+    } while (allChannels[indexAllChannels] !== undefined)
 
-    } else {
-        throw new Error(`Could not retrieve favorite channels (${respFavChannels.status}: ${respFavChannels.statusText})`);
-    }
-
-    if (parsedChannelMessages) {
-        dataStore.save("fav_chan_messages_realname", parsedChannelMessages)
-    }
+    dataStore.save("favorite_channels_members", parsedAllChannels)
+    dataStore.save("fav_chan_messages_realname", parsedChannelMessages)
 }
 
 async function fullSync({dataStore, client}) {
-    return changeIDbyRealName(dataStore, client, false);
+    let timestamp = moment().subtract(14, 'days').unix();
+    return changeIDbyRealName(dataStore, client, timestamp);
 }
 
 async function incrementalSync({dataStore, client, latestSynchronizationTime}) {
-    return changeIDbyRealName(dataStore, client, true, latestSynchronizationTime);
+    const lastSyncDateTime = new Date(latestSynchronizationTime)
+    const timestamp = `${lastSyncDateTime.getTime() / 1000}`.slice(0, 10);
```

Comment:
Can be simplified as in previous function:
```js
const timestamp = moment(latestSynchronizationTime).unix()
```

Reply:
According to local tests using this parameter `--since=2021-01-01T01:02:03Z` to run the test, `latestSynchronizationTime` receives a timestamp. 
And it causes this error when the code changes to your sugestion:
`TypeError: invokeMember (getTime) on java.time.Instant@2ecf5915 failed due to: Unknown identifier: getTime`
I'm assuming you are trying to sugest:
`const timestamp = moment(lastSyncDateTime).unix()`
insted of
`const timestamp = moment(latestSynchronizationTime).unix()`

Reply:
I don't understand the error message.
In my code snippet, I don't call `.getTime()` anywhere. Are you sure that you call it right?

```js
async function incrementalSync({dataStore, client, latestSynchronizationTime}) {
    const timestamp = moment(latestSynchronizationTime).unix();
    return changeIDbyRealName(dataStore, client, timestamp);
}
```

`moment()` accepts a string as a parameter.

Reply:
I'm facing the error. Can't we do in the way i sugested?

Reply:
Any news?

Reply:
Yes, of course.

Reply:
Will there be any other change then, or can I approve and merge?

Reply:
@cassiovanhelden ?

Reply:
I'm waiting for someone to approve, there is nothing else from us.

Reply:
Yeah the approval is on my team, just asking if we can proceed. Thanks